node/node_test.go

```diff
@@ -146,3 +146,122 @@ func TestService_Delete(t *testing.T) {
 		})
 	}
 }
+
+func TestService_DeleteAll(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                    string
+		preparePodServiceMockFn func(m *mock_node.MockPodService)
+		prepareFakeClientSetFn  func() *fake.Clientset
+		wantErr                 bool
+	}{
+		{
+			name: "delete all nodes and pods on nodes",
+			preparePodServiceMockFn: func(m *mock_node.MockPodService) {
+				m.EXPECT().List(gomock.Any()).Return(&corev1.PodList{
+					Items: []corev1.Pod{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "pod1",
+							},
+							Spec: corev1.PodSpec{
+								NodeName: "node1",
+							},
+						},
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "pod2",
+							},
+							Spec: corev1.PodSpec{
+								NodeName: "node1",
+							},
+						},
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "other-pod1",
+							},
+							Spec: corev1.PodSpec{
+								NodeName: "node2",
+							},
+						},
+					},
+				}, nil)
+				m.EXPECT().DeleteAll(gomock.Any()).Return(nil)
+			},
+			prepareFakeClientSetFn: func() *fake.Clientset {
+				c := fake.NewSimpleClientset()
+				// add test data.
+				c.CoreV1().Nodes().Create(context.Background(), &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "node1",
+					},
+				}, metav1.CreateOptions{})
+				return c
+			},
+			wantErr: false,
+		},
+		{
+			name: "fail to delete no nodes",
```

Comment:
Could you double-check? I think this name is incorrect.

```suggestion
			name: "fail if deleting all pods returns error",
```